controlsfx/src/main/java/impl/org/controlsfx/skin/RangeSliderSkin.java

```diff
@@ -142,290 +126,197 @@ public RangeSliderSkin(final RangeSlider rangeSlider) {
-                            new ParentTraversalEngine(
-                                    rangeSlider.getScene().getRoot(), algorithm)
-                            .select(highThumb, Direction.NEXT)
-                            .requestFocus();
-                        }
-                        event.consume();
-                    }
+            } else if (KeyCode.LEFT.equals(event.getCode()) || KeyCode.KP_LEFT.equals(event.getCode())) {
+                if (getSkinnable().getOrientation() == Orientation.HORIZONTAL) {
+                    rtl(getSkinnable(), this::incrementValue, this::decrementValue);
+                }
+            } else if (KeyCode.RIGHT.equals(event.getCode()) || KeyCode.KP_RIGHT.equals(event.getCode())) {
+                if (getSkinnable().getOrientation() == Orientation.HORIZONTAL) {
+                    rtl(getSkinnable(), this::decrementValue, this::incrementValue);
+                }
+            } else if (KeyCode.DOWN.equals(event.getCode()) || KeyCode.KP_DOWN.equals(event.getCode())) {
+                if (getSkinnable().getOrientation() == Orientation.VERTICAL) {
+                    decrementValue();
+                }
+            } else if (KeyCode.UP.equals(event.getCode()) || KeyCode.KP_UP.equals(event.getCode())) {
+                if (getSkinnable().getOrientation() == Orientation.VERTICAL) {
+                    incrementValue();
                 }
             }
+            event.consume();
         };
-        getSkinnable().addEventHandler(KeyEvent.KEY_PRESSED, keyEventHandler);  
-        // set up a callback on the behavior to indicate which thumb is currently 
-        // selected (via enum).
-        rangeSliderBehavior.setSelectedValue(new Callback<Void, FocusedChild>() {
-            @Override public FocusedChild call(Void v) {
-                return currentFocus;
+        EventHandler<KeyEvent> keyReleaseEventHandler = event -> {
+            if (KeyCode.HOME.equals(event.getCode())) {
+                home();
+            } else if (KeyCode.END.equals(event.getCode())) {
+                end();
             }
-        });
+            event.consume();
+        };
+        getSkinnable().addEventHandler(KeyEvent.KEY_PRESSED, keyPressEventHandler);
+        getSkinnable().addEventHandler(KeyEvent.KEY_RELEASED, keyReleaseEventHandler);
+        // set up a callback to indicate which thumb is currently selected (via enum).
+        setSelectedValue(v -> currentFocus);
+    }
+    
+    private void focusPreviousChild() {
+        final int index = getSkinnable().getParent().getChildrenUnmodifiable().indexOf(getSkinnable());
+        if (index != 0) {
+            getSkinnable().getParent().getChildrenUnmodifiable().get(index - 1).requestFocus();
+        } else {
+            getSkinnable().getParent().requestFocus();
+        }
+    }
+
+    private void focusNextChild() {
+        final int index = getSkinnable().getParent().getChildrenUnmodifiable().indexOf(getSkinnable());

```

Comment:
Would it be safer to assign getChildrenUnmodifiable to a local `final var` for reuse?